### PR TITLE
predownload image after some number updated pods ready

### LINKED ISSUE #876: [feature request] PreDownloadImage after some number count pods  update ready
**What would you like to be added**:
CloneSet trigger PreDownloadImage after some updated pods ready.

**Why is this needed**:
If the new revision image has some problem, predownload the image for the pods for inplace update is wasteful.

### PR DESCRIPTION
**Pull request by shiyan2016**
### Ⅰ. Describe what this PR does
predownload image after some number updated pods ready

### Ⅱ. Does this pull request fix one issue?

fixes #876

### CODE REVIEW

File: pkg/controller/cloneset/cloneset_controller.go [hunk 1]
```diff
@@ -301,9 +303,23 @@ func (r *ReconcileCloneSet) doReconcile(request reconcile.Request) (res reconcil
...
 	if !isPreDownloadDisabled {
 		if currentRevision.Name != updateRevision.Name {
-			// pre-download images for new revision
-			if err := r.createImagePullJobsForInPlaceUpdate(instance, currentRevision, updateRevision); err != nil {
-				klog.Errorf("Failed to create ImagePullJobs for %s: %v", request, err)
+			// get clone pre-download annotation
+			minUpdatedReadyPodsCount := 0
+			minUpdatedReadyPods, ok := instance.Annotations[appspub.ImagePreDownloadMinUpdatedReadyPods]
+			if ok {
+				if minUpdatedReadyPodsCount, err = strconv.Atoi(minUpdatedReadyPods); err != nil {
```
Comment (FillZpp) [inline pkg/controller/cloneset/cloneset_controller.go:310]: Should we use intOrString to support percentage or just integer?

Comment (shiyan2016) [reply pkg/controller/cloneset/cloneset_controller.go]: done

File: apis/apps/pub/image_predownload.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package pub
+
+const (
+	ImagePreDownloadMinUpdatedReadyPods = "apps.kruise.io/image-predownload-min-updated-ready-pods"
```
[the remark this answers is not in the document]
Comment (shiyan2016) [reply apis/apps/pub/image_predownload.go]: done

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**